python/cudf/cudf/core/indexed_frame.py

```diff
@@ -1569,18 +1570,32 @@ def drop_duplicates(
         keys = self._positions_from_column_names(
             subset, offset_by_index_columns=not ignore_index
         )
-        return self._from_columns_like_self(
-            libcudf.stream_compaction.drop_duplicates(
-                list(self._columns)
-                if ignore_index
-                else list(self._index._columns + self._columns),
-                keys=keys,
-                keep=keep,
-                nulls_are_equal=nulls_are_equal,
-            ),
-            self._column_names,
-            self._index.names if not ignore_index else None,
-        )
+        if preserve_order:
```

Comment:
`preserve_order` will just be passed into `libcudf.stream_compaction.drop_duplicates`.